File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_candies = 0;
        for(int i = 0;i < candies.size();i++){
            max_candies = max(max_candies, candies[i]);
        }
        vector<bool> result;
        for(int i = 0;i < candies.size();i++){
            if(candies[i] + extraCandies >= max_candies){
            result.push_back(true);
            }else{
            result.push_back(false);
            }
        }
        return result;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
          int maxCandies = 0; // camelCase
        
        for (int i = 0; i < candies.length; i++) {
            if (candies[i] > maxCandies) {
                maxCandies = candies[i];
            }
        }
        
        List<Boolean> result = new ArrayList<>(); // camelCase
        
        for (int i = 0; i < candies.length; i++) {
            // Smart shortcut using camelCase variables
            result.add(candies[i] + extraCandies >= maxCandies);
        }
        
        return result;
    }
}